### DISCUSSION TITLE
Forcing SSL for proxied containers

### DISCUSSION THREAD

**Discussion by marcmsk** [nginx-proxy/acme-companion#971] [Q&A]
Hi Folks, 

sorry if you seen this topic already, I not able to find any relevant threads.

Is it possible to force ssl at nginx-proxy for all app containers. I mean to translate or redirect external http requests for proxied containers to https, except for nginx-proxy-gen, nginx-proxy-acme. 

On a standard nginx, witthout a reverse proxy, I can specify in config for a domain to redirect http to https automatically. But I am not sure how to configure it behind nginx-proxy. Since internal communication between nginx-proxy and apps goes over 80, and config of ngnix-proxy is generated dynamically, also I believe acme needs http for comunication with Letsencrypt servers right?

I am currently migrating all my php apps to the dockerized environment. Before migrations I was testing all through http first, and since all was working I deployed also acme with nginx-gen. All was working well untill I started deployment of a new apps and noticed that pages are requested over https, but static content is blocked by browser becuase is called through http:

This is example of a fresh WordPress installation:
Blocked loading mixed active content “http://..../wp-admin/css/install.min.css?ver=6.1”
[setup-config.php](https://..../wp-admin/setup-config.php)
Blocked loading mixed active content “http://..../wp-includes/js/jquery/jquery.min.js?ver=3.6.1”
[setup-config.php](https://..../wp-admin/setup-config.php)
Loading failed for the <script> with source “http://.../wp-includes/js/jquery/jquery.min.js?ver=3.6.1”. [setup-config.php:32:1](https://..../wp-admin/setup-config.php)

Edit...
Ok I think that problem might be with the mginx.tmpl for nginx-proxy-gen, maybe a missing proxy_set_header, but I don't get that am the only one who has the issue.

**Comment by bplein** [comment]
I'm using nginx-proxy/nginx-proxy with nginx-proxy/acme-companion and it does this all automatically by default.

It attaches the following to all port 80 servers in default.conf: 
```
        location / {
                return 301 https://$host$request_uri;
        }
```

**Reply by marcmsk** [reply]
I realized that problem was happening only with wordpress and the following fixed it:
[https://www.linuxquestions.org/questions/blog/jdrosales-1161108/wordpress-behind-nginx-reverse-proxy-with-ssl-in-proxy-38464/](url)

It looks like wordpress developers don't care about nginx-proxy, and I have to keep editing core of wordpress after each update to avoid this behaviour.
